src/main/java/org/spongepowered/common/entity/player/SpongeUser.java

```diff
@@ -185,77 +185,137 @@ public DataContainer toContainer() {
 
     @Override
     public boolean canEquip(EquipmentType type) {
-        throw new UnsupportedOperationException(); // TODO Inventory API
+        if (this.self.isOnline()) {
```

Comment:
This should work, and make it cleaner:
`this.self.getPlayer().orElseThrow(() -> new UnsupportedOperationException("User is offline, offline inventory API yet to be implemented")).canEquip(type)`